### PR TITLE
[BEAM-2143]: Fix default temp location for DataflowRunner

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
- https://github.com/apache/beam/pull/2602 made `gs://some-bucket` path invalid and expects it to be of the form `gs://some-bucket/basename`. So adding `/temp` as the basename for the default `tempLocation`.
- Also fix log messages.

### CODE REVIEW

**Comment by vikkyrk** [comment]
R: @bjchambers @davorbonaci

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java [hunk 1]
```diff
@@ -117,10 +119,12 @@
    * Returns a default staging location under {@link GcpOptions#getGcpTempLocation}.
    */
   class StagingLocationFactory implements DefaultValueFactory<String> {
+    private static final Logger LOG = LoggerFactory.getLogger(StagingLocationFactory.class);
 
     @Override
     public String create(PipelineOptions options) {
       GcsOptions gcsOptions = options.as(GcsOptions.class);
+      LOG.info("No staging location provided, falling back to temp location.");
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java]: suggest formatting the options as options: "No stagingLocation provided, falling back to gcpTempLocation"

File: sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptions.java [hunk 2]
```diff
@@ -278,7 +278,7 @@ static String tryCreateDefaultBucket(
       }
       final String bucketName =
           "dataflow-staging-" + region + "-" + projectNumber;
-      LOG.info("No staging location provided, attempting to use default bucket: {}",
+      LOG.info("No temp location provided, attempting to use default bucket: {}",
```
Comment (bjchambers) [inline sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptions.java]: "No tempLocation specified, attempting to use default bucket:"

**Comment by vikkyrk** [comment]
If `tempLocation` is not specified:
  `INFO: No tempLocation specified, attempting to use default bucket: dataflow-staging-us-central1-996210899482`

If both `tempLocation` and `stagingLocation` is not specified
`INFO: No tempLocation specified, attempting to use default bucket: dataflow-staging-us-central1-996210899482`
`INFO: No stagingLocation provided, falling back to gcpTempLocation`

**Comment by vikkyrk** [comment]
Done

**Comment by bjchambers** [comment]
The exception I get from this is still not particularly useful. It says:
```
...
Caused by: java.lang.IllegalArgumentException: Missing object or bucket in path: 'gs://dataflow-staging-us-central1-927334603519/', did you mean: 'gs://some-bucket/dataflow-staging-us-central1-927334603519'?
...
```

**Comment by vikkyrk** [comment]
What command did you run? You shouldn't see an exception after this change for missing either of `tempLocation` or `stagingLocation`

**Comment by bjchambers** [comment]
Ah -- it looks like it wasn't using the patched version of the SDK. Looks good now.

**Comment by vikkyrk** [comment]
@bjchambers @davorbonaci Looks good to be merged.

**Pull request closed without merging**